src/lib/update-alternatives/model/alternative.rb

```diff
@@ -101,11 +104,22 @@ def choice(new_choice_path)
       end
       @value = new_choice_path
       @status = "manual"
+      @modified = true
     end
 
     def automatic_mode
       @status = "auto"
       @value = choices.sort_by { |choice| choice.priority.to_i }.last.path
+      @modified = true
+    end
+
+    def save
+      return unless @modified
+      if @status == "auto"
+        UpdateAlternatives::AutomaticModeCommand.new(self).execute
+      else
+        UpdateAlternatives::SetChoiceCommand.new(self).execute
+      end
```

Comment:
well, it is not enough robust for future changes...e.g. if new status appear...what is my prefered way to do it, is constant with exception when unknown value appear:

```
STATUS_COMMANDS = {
  "auto"      => UpdateAlternatives::AutomaticModeCommand,
  "manual" => UpdateAlternatives::SetChoiceCommand
}
STATUS_COMMANDS.default_proc = lambda { |_h, k| raise "unknown status '#{k}'" }

def save
  return unless @modified
  STATUS_COMMANDS[@status].new(self).execute
end
```
which maybe also make sense to consider to change API for command and have only single method `execute(alternative)` so `new(self).execute` is not needed, as it is its single responsibility.